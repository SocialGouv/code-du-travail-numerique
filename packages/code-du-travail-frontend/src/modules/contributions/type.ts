import {
  Breadcrumb,
  ContributionContentBase,
  ContributionConventionnelInfos,
  ContributionDocumentJson,
  ContributionFicheSpContent,
  ContributionGenericInfos,
  ContributionGenericNoCDTContent,
  ContributionHighlight,
  ContributionMetadata,
  DocumentElasticWithSource,
  ExportContributionFullLinkedContent,
} from "@socialgouv/cdtn-types";
import { RelatedItem } from "../documents";

type ExportContributionInfo = {
  breadcrumbs: Breadcrumb[];
  highlight?: ContributionHighlight;
  messageBlock?: string;
};

export type ContributionContent = Partial<ContributionContentBase> &
  Partial<ContributionFicheSpContent> &
  Partial<ContributionGenericNoCDTContent>;

type ContributionElasticDocumentBase = Omit<
  DocumentElasticWithSource<Omit<ContributionDocumentJson, "linkedContent">>,
  "breadcrumbs"
> &
  ContributionMetadata &
  ContributionContent &
  ExportContributionFullLinkedContent &
  ExportContributionInfo & {
    raw: string;
    url: string;
    content: string;
  };

export type ContributionElasticDocument = ContributionElasticDocumentBase &
  Partial<ContributionGenericInfos> &
  Partial<ContributionConventionnelInfos>;

export type ContributionRelatedItems = {
  title: string;
  items: RelatedItem[];
};

export type Contribution = ContributionElasticDocument & {
  isGeneric: boolean;
  isNoCDT: boolean;
  isFicheSP: boolean;
  relatedItems: ContributionRelatedItems[];
};
