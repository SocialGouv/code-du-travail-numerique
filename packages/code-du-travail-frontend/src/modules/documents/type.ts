import { SOURCES } from "@socialgouv/cdtn-utils";
import { DocumentElastic } from "@socialgouv/cdtn-types/build/elastic/common";

export type DocumentElasticResult<T> = T & {
  _id: string;
};

export const sources = [
  SOURCES.TOOLS,
  SOURCES.SHEET_SP,
  SOURCES.SHEET_MT,
  SOURCES.LETTERS,
  SOURCES.CONTRIBUTIONS,
  SOURCES.EXTERNALS,
  SOURCES.LABOUR_LAW,
  SOURCES.SHEET_MT_PAGE,
] as const;

export type Source = (typeof sources)[number];

export const isSource = (x: any): x is Source => sources.includes(x);

export type RelatedItem = Pick<DocumentElastic, "title"> & {
  source: Source;
  url: string;
};
