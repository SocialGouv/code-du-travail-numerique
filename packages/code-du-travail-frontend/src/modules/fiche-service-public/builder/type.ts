export type FicheSPDataText = { type: "text"; text: string };

export type FicheSPDataElementWithElementChildren =
  | FicheSPDataImage
  | FicheSPDataOuSAdresser
  | FicheSPDataWithElementChildren
  | FicheSPDataLienExterneCommente
  | FicheSPDataBlocCas
  | FicheSPDataChapitre
  | FicheSPDataListeSituations
  | FicheSPDataSituation
  | FicheSPDataAvertissement
  | FicheSPDataTableau;

export type FicheSPDataElement = {
  type: "element";
  children: FicheSPDataElement[] | FicheSPDataParagraphe[] | FicheSPDataText[];
  name:
    | "Introduction"
    | "Titre"
    | "Description"
    | "FragmentConditionne"
    | "SousChapitre"
    | "Expression"
    | "MiseEnEvidence"
    | "Valeur"
    | "Exposant"
    | "LienIntra"
    | "LienInterne"
    | "Texte"
    | "Complement";
};

export type FicheSPData =
  | FicheSPDataElement
  | FicheSPDataText
  | FicheSPDataBlocCas
  | FicheSPDataParagraphe
  | FicheSPDataLienExterneCommente
  | FicheSPDataLienExterne
  | FicheSPDataRessourceWeb
  | FicheSPDataOuSAdresser
  | FicheSPDataWithElementChildren
  | FicheSPDataSituation
  | FicheSPDataListeSituations
  | FicheSPDataImage
  | FicheSPDataAvertissement
  | FicheSPDataList
  | FicheSPDataServiceEnLigne
  | FicheSPDataChapitre
  | FicheSPDataTableau
  | FicheSPDataTableauChildren
  | FicheSPDataTitreFlottant;

export type FicheSPDataBlocCas = {
  type: "element";
  attributes: {
    affichage: string;
  };
  children: FicheSPDataWithElementChildren[];
  name: "BlocCas";
};

export type FicheSPDataParagraphe = {
  type: "element";
  children: FicheSPDataText[];
  name: "Paragraphe";
};
export type FicheSPDataWithElementChildren = {
  type: "element";
  children: FicheSPDataElement[];
  name: "ANoter" | "ASavoir" | "Rappel" | "Cas" | "Exemple";
};
export type FicheSPDataTableau = {
  type: "element";
  children: FicheSPDataTableauChildren[];
  name: "Tableau";
};
export type FicheSPDataTableauChildren = {
  type: "element";
  attributes: {
    type: string;
  };
  children: FicheSPDataElement[];
  name: "Rang√©e" | "Colonne";
};

export type FicheSPDataChapitre = {
  type: "element";
  children: FicheSPDataElement[] | FicheSPDataParagraphe[];
  name: "Chapitre";
};

export type FicheSPDataAvertissement = {
  type: "element";
  attributes?: { date: string };
  children: FicheSPDataElement[];
  name: "Avertissement" | "Attention";
};

export type FicheSPDataImage = {
  type: "element";
  attributes: {
    LienPublication: string;
    langue: string;
    poids: string;
    format: string;
    type: string;
    redimensionnable: string;
  };
  children: FicheSPDataElement[];
  name: "Image";
};

export type FicheSPDataLienExterneCommente = {
  type: "element";
  attributes: {
    URL: string;
  };
  children: FicheSPDataElement[];
  name: "LienExterneCommente";
};

export type FicheSPDataLienExterne = {
  type: "element";
  attributes: {
    URL: string;
  };
  children: FicheSPDataText[];
  name: "LienExterne";
};

export type FicheSPDataList = {
  type: "element";
  attributes: { type: string };
  children: FicheSPDataElement[];
  name: "Liste";
};

export type FicheSPDataRessourceWeb = {
  type: "element";
  attributes: {
    URL: string;
  };
  children: FicheSPDataElement[];
  name: "RessourceWeb";
};

export type FicheSPDataOuSAdresser = {
  type: "element";
  children: (FicheSPDataRessourceWeb | FicheSPDataElement)[];
  name: "OuSAdresser";
};

export type FicheSPDataServiceEnLigne = {
  type: "element";
  attributes: { URL: string };
  children: FicheSPDataElement[];
  name: "ServiceEnLigne" | "PourEnSavoirPlus";
};

export type FicheSPDataSituation = {
  type: "element";
  children: FicheSPDataElement[];
  name: "Situation";
};

export type FicheSPDataListeSituations = {
  type: "element";
  attributes: {
    affichage: string;
  };
  children: FicheSPDataSituation[];
  name: "ListeSituations";
};

export type FicheSPDataTitreFlottant = {
  type: "element";
  children: FicheSPDataParagraphe[];
  name: "TitreFlottant";
};
