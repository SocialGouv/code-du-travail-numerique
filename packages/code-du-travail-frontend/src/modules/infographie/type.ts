// A remplacer par les type du package cdtn-type

import { SOURCES } from "@socialgouv/cdtn-utils";
import { Breadcrumb, DocumentElasticWithSource } from "@socialgouv/cdtn-types";

export type InfographicElasticDocument = DocumentElasticWithSource<
  InfographicTemplateDoc,
  typeof SOURCES.INFOGRAPHICS
>;

export type InfographicTemplateDoc = {
  meta_title: string;
  date: string;
  author: string;
  svgFilename: string;
  svgFilesizeOctet: number;
  pdfFilename: string;
  pdfFilesizeOctet: number;
  description: string;
  meta_description: string;
  transcription: string;
};

export type Infographic = Pick<
  InfographicElasticDocument,
  | "title"
  | "meta_title"
  | "description"
  | "meta_description"
  | "date"
  | "transcription"
> & {
  svgUrl: string;
  pdf: {
    url: string;
    sizeOctet: string;
  };
  breadcrumbs: Breadcrumb[];
};
