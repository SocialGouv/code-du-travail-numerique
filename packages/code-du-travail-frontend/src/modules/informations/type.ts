import {
  DocumentElasticWithSource,
  EditorialContentDoc,
} from "@socialgouv/cdtn-types";
import { SOURCES } from "@socialgouv/cdtn-utils";

export type EditorialContentElasticDocument = Omit<
  DocumentElasticWithSource<
    KeysToCamelCase<EditorialContentDoc>,
    typeof SOURCES.EDITORIAL_CONTENT
  >,
  "introWithGlossary"
>;

type Camelize<T extends string> = T extends `${infer A}_${infer B}`
  ? `${A}${Camelize<Capitalize<B>>}`
  : T;

export type KeysToCamelCase<T> = {
  [K in keyof T as Camelize<string & K>]: T[K] extends (infer U)[]
    ? KeysToCamelCase<U>[]
    : T[K] extends {}
      ? KeysToCamelCase<T[K]>
      : T[K];
};
