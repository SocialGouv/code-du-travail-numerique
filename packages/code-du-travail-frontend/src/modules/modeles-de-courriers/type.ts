import {
  DocumentElasticWithSource,
  MailTemplateDoc,
} from "@socialgouv/cdtn-types";

export type ElasticLaborMailTemplateDoc =
  DocumentElasticWithSource<MailTemplateDoc>;
