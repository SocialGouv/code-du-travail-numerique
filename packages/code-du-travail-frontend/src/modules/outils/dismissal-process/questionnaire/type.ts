export type PreviousResponse = {
  index: number;
  text?: string;
  info?: string;
};

export type SlugResponses = {
  [slug: string]: PreviousResponse[];
};

export type QuestionnaireResponse = {
  text: string;
  trackingName: string;
  info?: string;
  description?: string;
  slug?: string;
  statement: string;
  neutralStatement?: string;
  neutralStatementRef?: boolean;
  neutralStatementInfo?: string;
  question?: QuestionnaireQuestion;
};

export type QuestionnaireQuestion = {
  text: string;
  trackingName: string;
  info?: string;
  description?: string;
  responses: QuestionnaireResponse[];
};
