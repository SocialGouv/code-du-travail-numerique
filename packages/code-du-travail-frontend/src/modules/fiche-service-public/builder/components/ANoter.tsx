import React, { JSX } from "react";
import { ElementBuilder } from "./ElementBuilder";
import { fr } from "@codegouvfr/react-dsfr";
import { filterOutTitle, getInChildrenByName, getText } from "../utils";
import { FicheSPDataElementWithElementChildren } from "../type";

export const ANoter = ({
  data,
  headingLevel,
}: {
  data: FicheSPDataElementWithElementChildren;
  headingLevel: number;
}) => {
  const title = getInChildrenByName(data, "Titre");
  const Heading = `h${headingLevel + 2}` as keyof JSX.IntrinsicElements;
  return (
    <div className={fr.cx("fr-alert", "fr-alert--info", "fr-mb-4w")}>
      {title && <Heading>{getText(title)}</Heading>}
      <ElementBuilder
        data={filterOutTitle(data)}
        headingLevel={headingLevel + 1}
      />
    </div>
  );
};

export default ANoter;
