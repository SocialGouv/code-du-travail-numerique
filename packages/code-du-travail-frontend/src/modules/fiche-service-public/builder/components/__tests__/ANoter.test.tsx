import React from "react";
import { render } from "@testing-library/react";
import data from "./mocks/aNoterData.json";
import ANoter from "../ANoter";
import { FicheSPDataElementWithElementChildren } from "../../type";

describe("<ANoter />", () => {
  it("affiche un encart Ã  noter (alert)", () => {
    const { container } = render(
      <ANoter
        data={data as FicheSPDataElementWithElementChildren}
        headingLevel={0}
      />
    );
    expect(container).toMatchSnapshot();
    expect(container.children[0]).toHaveClass("fr-alert--info");
  });
});
