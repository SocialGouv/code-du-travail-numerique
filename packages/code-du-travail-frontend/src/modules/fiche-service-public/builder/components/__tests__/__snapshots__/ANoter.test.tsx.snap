// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<ANoter /> affiche un encart à noter (alert) 1`] = `
<div>
  <div
    class="fr-alert fr-alert--info fr-mb-4w"
  >
    <h2>
      À savoir
    </h2>
    <p>
      le préavis commence à courir à compter de la notification à l'employeur de votre démission.
    </p>
  </div>
</div>
`;
